docs/rules/object-curly-newline.md

```diff
@@ -19,6 +19,7 @@ Or an object option:
 
 * `"multiline": true` (default) requires line breaks if there are line breaks inside properties or between properties
 * `"minProperties"` requires line breaks if the number of properties is more than the given integer
+* `"consistent"` requires that either both curly braces, or neither, directly enclose newlines
```

Comment:
We should make sure to clearly document this behavior. For example:

> * `"minProperties"` requires line breaks if the number of properties is more than the given integer. By default, an error will also be reported if an object contains linebreaks and has fewer properties than the given integer. However, the second behavior is disabled if the `consistent` option is set to `true`.

I don't think this is a very intuitive behavior. It would be better to come up with something that wouldn't implicitly change the behavior of `minProperties` when `consistent` is true.